src/xenia/kernel/kernel_state.cc

```diff
@@ -316,6 +316,23 @@ void KernelState::SetExecutableModule(object_ref<UserModule> module) {
     *variable_ptr = executable_module_->hmodule_ptr();
   }
 
+  // Setup the kernel's ExLoadedImageName field
+  export_entry = processor()->export_resolver()->GetExportByOrdinal(
+      "xboxkrnl.exe", ordinals::ExLoadedImageName);
+
+  if (export_entry) {
+    char* variable_ptr =
+        memory()->TranslateVirtual<char*>(export_entry->variable_ptr);
+    size_t length = 0;
+
+    if (module) {
+      std::string image_name(module->path());
+      length = std::min(size_t(FILENAME_MAX - 1), image_name.size());
```

Comment:
That should be the guest maximum file name size since this structure writes to the guest memory, not the host one.